sympy/integrals/integrals.py

```diff
@@ -1160,8 +1159,50 @@ def _eval_integral(self, f, x, meijerg=None, risch=None, manual=None,
                 parts.append(coeff * h)
             else:
                 return None
-
-        return Add(*parts)
+            current_expr = f
+            for lim in self.limits:
+                if len(lim) == 3:
```

Comment:
This function is already far too long and messy. I would rather not extend it like this.

Instead there can be a new function for this.